src/collapse/common.rs

```diff
@@ -0,0 +1,794 @@
+use std::borrow::Cow;
+use std::io;
+use std::mem;
+use std::sync::Arc;
+
+use chashmap::CHashMap;
+use crossbeam::channel;
+use fnv::FnvHashMap;
+use lazy_static::lazy_static;
+
+const CAPACITY_HASHMAP: usize = 512;
+pub(crate) const CAPACITY_READER: usize = 128 * 1024;
+pub(crate) const DEFAULT_NSTACKS_PER_JOB: usize = 100;
+const NBYTES_PER_STACK_GUESS: usize = 1024;
+const RUST_HASH_LENGTH: usize = 17;
```

Comment:
I'd love to see brief docs for how these constants were chosen.

Reply:
You'll have to fill in `CAPACITY_HASHMAP`, `CAPACITY_READER`, and `RUST_HASH_LENGTH`, as I don't know where those came from.

I'll take a crack at commenting `DEFAULT_NSTACKS_PER_JOB` and `NBYTES_PER_STACK_GUESS`.

Reply:
That's fine -- we can update those later :)